### PR TITLE
new nav and header

### PR DESCRIPTION
**Pull request by spasovski**
Changes everything so too many screenshots to post. I am not happy with where the new styles sit so they are likely to be moved to different/new files.

TODO:
- Fix tests!
- Add the new "popular/new" toggle.

### CODE REVIEW

File: src/media/img/marketplace_logo.svg [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<svg width="166px" height="32px" viewBox="0 0 166 32" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:sketch="http://www.bohemiancoding.com/sketch/ns">
+    <!-- Generator: Sketch 3.3.3 (12081) - http://www.bohemiancoding.com/sketch -->
+    <title>Shape + Shape + Shape + Imported Layers</title>
```
Comment (diox) [inline src/media/img/marketplace_logo.svg]: Can we optimize that svg ? We probably don't need the medata and the ids.

Comment (diox) [inline src/media/img/marketplace_logo.svg]: Same for the other svgs

**Comment by ngokevin** [comment]
Here are the [Sherlocked captures](http://sherlocked.dev.mozaws.net/builds/86549641) before it crapped out on Collection on Mobile.

**Comment by ngokevin** [comment]
Search broken on Desktop Firefox.

screen shot 2015-10-21 at 10 45 01 am

**Comment by ngokevin** [comment]
Wordmark disppearing on Desktop Firefox.

screen shot 2015-10-21 at 10 46 07 am

**Comment by ngokevin** [comment]
No hover styles on Desktop Firefox

screen shot 2015-10-21 at 10 46 48 am

**Comment by ngokevin** [comment]
Header bar is empty on most pages

screen shot 2015-10-21 at 10 48 25 am

**Comment by ngokevin** [comment]
Tabs cut off in the Firefox OS simulator

screen shot 2015-10-21 at 10 50 34 am

**Comment by ngokevin** [comment]
Scrolling through categories is causing the page to scroll rather than the category list. Also these look like small click targets, could use some more padding (Weather looks hard to click).

screen shot 2015-10-21 at 10 51 40 am

**Comment by ngokevin** [comment]
Blue dot that will for sure drive OCD-inclined people crazy

screen shot 2015-10-21 at 10 53 40 am

**Comment by ngokevin** [comment]
@spasovski help me click around and QA your stuff, test multiple viewports, states, all the pages.

**Comment by ngokevin** [comment]
User dropdown is broken

**Comment by ngokevin** [comment]
Let me know when you've tested your PR more + fixed issues.

**Comment by ngokevin** [comment]
btw i'm glad this is pretty close, the nav itself is pretty functional

**Comment by ngokevin** [comment]
Also of importance, the Add-on nav item should be hidden on desktop Firefox using core/capabilities.

**Comment by diox** [comment]
> Also of importance, the Add-on nav item should be hidden on desktop Firefox using core/capabilities.

Just `settings.addonsEnabled` should be enough now that the package sets it. We should just never set it to true in settings - alternatively, change the way the package is transmitting the info.

**Comment by ngokevin** [comment]
Will start reviewing this. In the mean time you can delete failing tests related to the old nav.

**Comment by ngokevin** [comment]
Went to a category:

screen shot 2015-10-22 at 11 33 20 am

**Comment by ngokevin** [comment]
The blue dot is still there

screen shot 2015-10-22 at 11 34 22 am

**Comment by ngokevin** [comment]
Huge delay clicking the user icon and seeing the dropdown:

screen shot 2015-10-22 at 11 34 46 am

**Comment by ngokevin** [comment]
Big delay and flickers when clicking search as well on desktop

**Comment by ngokevin** [comment]
doesn't look right

screen shot 2015-10-22 at 11 35 55 am

**Comment by ngokevin** [comment]
user nav doesn't go away when navigating

screen shot 2015-10-22 at 11 36 19 am

**Comment by ngokevin** [comment]
click around more and lmk when you addressed + tested more. do you see this locally?

**Comment by spasovski** [comment]
yikes, having trouble logging in locally so testing the settings menu is a pain but still doable

**Comment by ngokevin** [comment]
should work out of the box if you do the frontend setup. might be a pain but still have to test it.

**Comment by ngokevin** [comment]
i thought we were going to show stuff in the header

screen shot 2015-10-22 at 4 11 26 pm

**Comment by ngokevin** [comment]
dropdowns no longer slide down / lost their transitions

desktop search has a white underline

screen shot 2015-10-22 at 4 12 15 pm

**Comment by ngokevin** [comment]
on desktop. clicking search should de-toggle categories dropdown. clicking categories should de-toggle search dropdown

**Comment by ngokevin** [comment]
home not highlighted 

screen shot 2015-10-22 at 4 14 39 pm

**Comment by ngokevin** [comment]
platform selector is broken

screen shot 2015-10-22 at 4 15 33 pm

**Comment by ngokevin** [comment]
icon sizes different

screen shot 2015-10-22 at 4 16 09 pm

**Comment by ngokevin** [comment]
search isn't focused on click

screen shot 2015-10-22 at 4 17 33 pm

**Comment by ngokevin** [comment]
contextual nav not working perfectly. Apps should be highlighted when on an app-related page:

screen shot 2015-10-22 at 4 20 04 pm

**Comment by ngokevin** [comment]
Feed a bit hard to re-discover when navigating away from it, but that's probably ok.

**Comment by ngokevin** [comment]
we don't need this content filter anymore

screen shot 2015-10-22 at 4 22 51 pm

**Comment by ngokevin** [comment]
big gap 

screen shot 2015-10-22 at 4 24 15 pm

File: src/media/css/nav--global.styl [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+}
+
+// TODO: Define this breakpoint.
+@media (min-width: 400px) {
+    .item-homescreens {
+        display: block;
+    }
+}
+
+@media $base-desktop {
```
Comment (ngokevin) [inline src/media/css/nav--global.styl]: use the +media--base-desktop() pattern

**Comment by krupa** [comment]
https://www.dropbox.com/s/ybwwc4wikq70a55/Screenshot%202015-10-22%2018.29.23.png?dl=0
* The hamburger icon(?) needs fixing
* Since we have content_type in the header, we can get rid of the dropdwon box to switch between apps and websites.

Also, on the phone when accessing via the browser, the footer is hidden by default until the user scrolls down. We need to fix that.

Clicking the search icon, pushes me to the category landing page for Music? I have to click twice to get to the Search field.

Searching for add-ons doesn't work.

**Comment by chuckharmston** [comment]
screen shot 2015-10-22 at 7 25 56 pm

- [ ] Hamburger icon is funky.
- [ ] There's no hamburger icon hover state.
- [ ] The search icon is far too big.
- [ ] The arrow should be vertically centered in the middle of the lowercase arrows.
- [ ] There should be space between the text and the arrow..

**Comment by chuckharmston** [comment]
screen shot 2015-10-22 at 7 38 39 pm

The hamburger and search icons should be the same size, and occupy identical space on either side of the category dropdown.

**Comment by chuckharmston** [comment]
screen shot 2015-10-22 at 7 40 24 pm

- [ ] I don't have mocks to check, but I'm almost 100% sure that we don't want a gradient shadow on our almost-entirely flat design.
- [ ] There should be margin between each nav item equal to the horizontal padding on `.global-nav-menu`

**Comment by chuckharmston** [comment]
screen shot 2015-10-22 at 7 43 07 pm

- [ ] The animation speed on the reveal is too slow.
- [ ] The ease feels wrong. I'm willing to bed that it doesn't match what we have elsewhere on the site. Try to be consistent.
- [ ] The left side of the Marketplace logo should be flush with the left text of the icons.
- [ ] I'm pretty sure there needs to be more vertical spacing on each side of the logo. It is very tight.
- [ ] The padding on `.full-screen-overlay` feels tight as well.

**Comment by ngokevin** [comment]
Btw the hamburger icon might be an artifact of my deployed setup don't know why

**Comment by chuckharmston** [comment]
screen shot 2015-10-22 at 7 49 45 pm

- [ ] There are no hover states on anything.
- [ ] Animation speeds are too slow; see previous comment.
- [ ] Arrow positioning is off; see previous comment.
- [ ] Padding on `.full-screen-overlay` should match whatever is done for the `...` menu. It feels too tight.
- [ ] The header size feels too small relative to the icons. I know you're trying to match what it is in the header of the unrevealed state, but the context is totally different.

**Comment by ngokevin** [comment]
@spasovski can you attach mocks please?

**Comment by chuckharmston** [comment]
I don't like the `All apps`. It doesn't indicate that it's a category menu.

**Comment by chuckharmston** [comment]
Make sure scrolldown on all pages accounts for the new vertical space taken up by the tab bar:

screen shot 2015-10-22 at 7 55 15 pm

screen shot 2015-10-22 at 7 55 01 pm

**Comment by chuckharmston** [comment]
I'm pretty sure we're supposed to have Newsletter, TOS, and Privacy Policy links in the more menu.

**Comment by chuckharmston** [comment]
Let's `display:none` the menus when they are off screen. It will make scrolling faster, redraws smoother, and prevent this ugliness:

screen recording 2015-10-22 at 07 59 pm

**Comment by chuckharmston** [comment]
Animation speed/easing and arrow positioning should roughly match this:

screen recording 2015-10-22 at 08 06 pm

**Comment by chuckharmston** [comment]
screen shot 2015-10-22 at 8 08 59 pm

We were supposed to carry through the hub/spoke headers, right?

**Comment by chuckharmston** [comment]
screen recording 2015-10-22 at 08 12 pm

Desktop search reveal is awkward and the visuals aren't correct. This shouldn't change at all.

**Comment by chuckharmston** [comment]
screen recording 2015-10-22 at 08 14 pm

- [ ] These hover states are weird. The one on "All Apps" is so subtle it looks wrong.
- [ ] Why is the text "All Apps" here at all on desktop?

**Comment by chuckharmston** [comment]
screen shot 2015-10-22 at 8 16 25 pm

- [ ] The whitespace here is weird. Let's add more top margin to the list.
- [ ] The max-width for this is far too big. Let's add `.addon-list { margin-left: auto; margin-right: auto; max-width 800px; }`

**Comment by chuckharmston** [comment]
Let's add a max-width to this, too. That button is too big even for your thumbs.

screen shot 2015-10-22 at 8 19 25 pm

**Comment by ngokevin** [comment]
nit: the two arrows are different and they're next to each other

screen shot 2015-10-26 at 10 56 41 am

**Comment by spasovski** [comment]
yea i am tempted to just use the non-sprited one (sharp corners)

**Comment by ngokevin** [comment]
When you log in, I see two headers.

**Comment by chuckharmston** [comment]
Video of it: http://chuck.mn/340n2m252Y03

**Comment by ngokevin** [comment]
The back button does not work as expected. We should not see it on root pages.

**Comment by ngokevin** [comment]
@pwalm says we're keeping hub+spoke headers

**Comment by ngokevin** [comment]
looks like add-ons have no name in search results, might be for a different PR 

screen shot 2015-10-26 at 2 02 41 pm

**Comment by ngokevin** [comment]
Not a big issue, but I can sometimes scroll the page through the More nav. Wait for it:

scroll

**Comment by ngokevin** [comment]
Minor issue, I can't see the placeholder despite not having set a value in the text box. The placeholder does flash momentarily.

screen shot 2015-10-26 at 2 09 23 pm

**Comment by ngokevin** [comment]
minor issue. the all apps hover state doesn't have enough padding. the current category is not highlighted.

hovers

**Comment by spasovski** [comment]
placeholder issue is due to `.focus()` which feels important for the mobile keyboard - treating as minor also

**Comment by ngokevin** [comment]
logged out state shows logout link

screen shot 2015-10-26 at 3 25 45 pm

**Comment by ngokevin** [comment]
minor request: possible to reset scroll state of the More nav when it's closed?

**Comment by ngokevin** [comment]
home should be highlighted

screen shot 2015-10-26 at 3 30 20 pm

**Pull request merged**